include/experimental/__p0009_bits/extents.hpp

```diff
@@ -337,12 +419,16 @@ class extents
   MDSPAN_INLINE_FUNCTION
   static constexpr
   size_type static_extent(size_t n) noexcept {
+    // Can't do assert here since that breaks true constexpr ness
+    // assert(n<rank());
```

Comment:
Should we consider using `is_constant_evaluated()` here?
```c++
if (! std::is_constant_evaluated()) {
  assert(n < rank());
}
```
See e.g., https://godbolt.org/z/hYKT5c9qM
```c++
#include <cassert>
#include <iostream>
#include <random>
#include <type_traits>

int random_number()
{
    // Seed with a real random value, if available
    std::random_device r;
 
    // Choose a random mean between 1 and 6
    std::default_random_engine e1(r());
    std::uniform_int_distribution<int> uniform_dist(1, 6);
    int mean = uniform_dist(e1);
    return mean;
}

constexpr int foo(int k) {
    if (std::is_constant_evaluated()) {
        return -1;
    } else {
        assert(k + 1 > 0);
        return k - 2;
    }
}

template<int k = foo(50)>
struct Thing {
  Thing() { std::cout << "Thing: k = " << k << std::endl; }
};

int main() {
  const int i = random_number();
  std::cout << "foo(" << i << ") = " << foo(i) << std::endl;
  std::cout << "foo(" << 42 << ") = " << foo(42) << std::endl;
  Thing<> thing_one;
  return 0;
}
```

Reply:
yeah we should try that. Just need to come up with a way in the constructors to actually spell the whole condition. But we can do that in another PR later.